### PR TITLE
O3-1309: Implement the search definitions feature

### PR DESCRIPTION
**Pull request by anjula-sack**
## Purpose

- The purpose of this PR is to fix #[O3-1309](https://issues.openmrs.org/browse/O3-1309)

## Goals

- To give the user the ability to search for saved cohorts and queries

## Approach

- Created the saved cohort and saved query components
- Added styles to match the design
- Wrote tests
- Updated the translations

## Screenshots
Screenshot 2022-08-02 at 10 33 25 AM
Screenshot 2022-08-02 at 10 33 40 AM

Screenshot 2022-07-31 at 9 43 27 PM

### CODE REVIEW

File: src/components/saved-cohorts/saved-cohorts.component.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+import mainStyles from "../../cohort-builder.scss";
+import { PaginationData } from "../../types";
+import EmptyData from "../empty-data/empty-data.component";
+import SavedCohortsOptions from "./saved-cohorts-options/saved-cohorts-options.component";
+import { deleteCohort, getCohorts } from "./saved-cohorts.resource";
+import styles from "./saved-cohorts.scss";
+
+interface SavedCohortsProps {
+  viewCohort: (queryId: string) => Promise<void>;
```
Comment (jayasanka-sack) [inline src/components/saved-cohorts/saved-cohorts.component.tsx:27]: Shall we use a proper naming convention for handlers?

ex: `onViewCohort`

File: src/components/saved-cohorts/saved-cohorts.component.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+}
+
+const SavedCohorts: React.FC<SavedCohortsProps> = ({ viewCohort }) => {
+  const [page, setPage] = useState(1);
```
Comment (jayasanka-sack) [inline src/components/saved-cohorts/saved-cohorts.component.tsx:31]: how about `currentPage` or `selectedPage`?

Comment (anjula-sack) [reply src/components/saved-cohorts/saved-cohorts.component.tsx:31]: this cannot be changed. It creates a conflict with the carbon pagination component pagination type

**Review by jayasanka-sack — APPROVED** [review]
Thanks @anjula-sack !

**Pull request merged**